hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestTableSnapshotScanner.java

```diff
@@ -190,6 +195,36 @@ public void testNoDuplicateResultsWhenSplitting() throws Exception {
     }
   }
 
+
+  @Test
+  public void testScanLimit() throws Exception {
+    setupCluster();
+    final TableName tableName = TableName.valueOf(name.getMethodName());
+    final String snapshotName = tableName + "Snapshot";
+    try {
+      createTableAndSnapshot(UTIL, tableName, snapshotName, 50);
+      Path restoreDir = UTIL.getDataTestDirOnTestFS(snapshotName);
+      Scan scan = new Scan().withStartRow(bbb).setLimit(100); // limit the scan
+
+      TableSnapshotScanner scanner =
+          new TableSnapshotScanner(UTIL.getConfiguration(), restoreDir, snapshotName, scan);
+      int count = 0;
+      while (true) {
+        Result result = scanner.next();
+        if (result == null) {
+          break;
+        }
+        count++;
+      }
+      Assert.assertEquals(100, count);
+      scanner.close();
```

Comment:
Can we move `scanner.close()` in finally? We can definte `TableSnapshotScanner scanner=null` outside of try {}, and in finally, we can check if scanner is not null, then close it.

Reply:
yes,You are quite right